ast/dml.go

```diff
@@ -1067,52 +1073,64 @@ func (n *UnionSelectList) Accept(v Visitor) (Node, bool) {
 	return v.Leave(n)
 }
 
-// UnionStmt represents "union statement"
+type SetOprType uint8
+
+const (
+	Union SetOprType = iota
+	UnionAll
+	Except
+	Intersect
+)
+
+// SetOprStmt represents "union/except/intersect statement"
 // See https://dev.mysql.com/doc/refman/5.7/en/union.html
-type UnionStmt struct {
+// See https://mariadb.com/kb/en/intersect/
+// See https://mariadb.com/kb/en/except/
+type SetOprStmt struct {
 	dmlNode
 	resultSetNode
 
-	SelectList *UnionSelectList
+	OptType    SetOprType
```

Comment:
```suggestion
	OprType    SetOprType
```
?

IMO it's clearer to spell it out as `SetOperator` than `SetOprType`.